IssuNumber : #2 2-usercrud

leebeanbin opened the pull request:
## 📌 연관된 이슈
<!-- 수정하려는 현재 동작을 설명하거나 관련 문제에 대한 링크를 제공해주세요 -->
> Issue Number : #이슈번호

## PR 종류
해당 PR은 어떤 부분에 대한 변화인가요?
<!-- Please check the one that applies to this PR using "x". -->
- [ ] 버그 수정 `FIX`(Hotfix)
- [x] 코드 수정 `MODIFY`
- [x] 기능 추가 `FEAT`
- [ ] 파일 이름 변경 `RENAME`
- [ ] 테스트 코드 추가 `TEST`
- [ ] 코드 스타일 변화 (formatting, local variables) `STYLE`
- [ ] 리팩토링 (no functional changes, no api changes) `REFACTOR`
- [ ] 빌드 관련 내용 추가 및 변화 `BUILD`
- [ ] 문서 내용 추가 및 변화`DOCS`
- [ ] 기타 변경 사항 추 `CHORE`
- [ ] Other... (적어주세요 : )

## 💻 작업 내용
> 이번 PR에서 작업한 내용을 간략히 설명해 주세요 (이미지 첨부 가능)

## 해당 PR이 다른 영역에 영향을 미치나요?
- [ ] Yes
- [ ] No

<!-- "Yes"를 선택한 경우, 어떤 영향을 끼치는지, 발생 위치 Path를 설명해주세요. -->

## 💬 리뷰 요구사항
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
> - ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

leebeanbin committed:
user signup&out 기본 세팅

jwt 인증 인가를 하지 않는 상태의 유저 회원가입 및 탈퇴의 로직을 구현하기 위한 기초 파일 구성

leebeanbin committed:
user signup&out 구현

현재 signout에서 softDelete가 제대로 이뤄지지 않는 부분을 수정

leebeanbin committed:
user signup&out 구현

현재 signout에서 softDelete가 제대로 이뤄지지 않는 부분 완료

leebeanbin committed:
user signup&out 구현

현재 signout에서 softDelete와 조회 기능 구현 완료

leebeanbin committed:
exception handler 적용

exception handler를 적용하여 오류 메세지 200과 400으로 통일

leebeanbin committed:
file location shifting

leebeanbin committed:
exception handler refactoring

crud에서 try-catch를 제외하고 exception handler에서 관리하도록 수정

leebeanbin committed:
exception handler refactoring

crud에서 try-catch를 제외하고 exception handler에서 관리하도록 수정

CKM123423 commented on Areadevelopment/src/main/java/com/sparta/areadevelopment/service/UserService.java:
```diff
@@ -17,20 +18,46 @@
+    }
+
+    private void checkForDuplicate(String username, String nickname, String email) {
+        // 하나의 쿼리로 중복 여부 확인
```
이메서드에서 쓰로우 날릴때 이걸 리스트로받아서 오류문 전부를 출력하는 방법이 없을까요?

지금 현재는 만약 유저네임, 이메일이 중복인상태라면

유저네임에서 익셉션이 발생하고 그게 글로벌핸들러로 타서 밑에 이메일 검증 익셉션은발생하지 않아 한번에 오류메세지로 모든 오류를 찾을수 없습니다

leebeanbin replied:
한번에 검증해서, 메세지를 쌓아 두고 출력하는 방식으로 다시 리펙토링 했습니다!

leebeanbin committed:
pr test

leebeanbin committed:
Exception handler refactoring

검증 로직을 전부 거쳐 오류가 발생한 부분을 전부 탐색 후 내용을 한번에 JSON 형식으로 출력한다. 또한 잘못된 URL을 입력했을 경우의 처리도 완료.

leebeanbin committed:
구글 컨벤션 재 적용 및 optimizing

leebeanbin committed:
튜터님의 수정 사항 반영 리펙토링

핸들러, 메서드 명, 리턴 값 조정

leebeanbin committed:
Merge branch 'dev' into 2-usercrud
